src/provider/controller/ClientEventController.ts

```diff
@@ -28,10 +29,10 @@ export class ClientEventController {
                     type: 'notification-created',
                     notification: mutable(notification)
                 };
-                eventPump.push<NotificationCreatedEvent>(source.uuid, event);
+                await eventPump.push<NotificationCreatedEvent>(source.uuid, event);
             } else if (action instanceof RemoveNotifications) {
                 const {notifications} = action;
-                notifications.forEach((storedNotification: StoredNotification) => {
+                await Promise.all(notifications.map(async (storedNotification: StoredNotification) => {
```

Comment:
This makes the order `NotificationClosedEvent`s fire unpredictable. Happy for this to be the case?

Reply:
The order the client app will recieved closed events from notifications is unpredictable anyway.
If `RemoveNotifications` is called by the UI via clearing a group, the notifications will be ordered by date as they are sorted.
If `RemoveNotifications` is called via `clearAll` they will be ordered by creation order (they are just pushed onto the array when created).